server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -91,63 +118,249 @@ private void registerListeners() {
+         * If the task has already started, then the current execution
+         * will not be interrupted but the task will not be scheduled anymore.
+         *
+         * @return false, if the task was already cancelled; {@code true}
+         * otherwise.
+         */
+        @Override
+        public boolean cancel() {
+            final boolean alreadyCancelled = cancelled;
+            cancelled = true;
+            if (alreadyCancelled == false) {
+                scheduledTask.cancel();
+            }
+            return alreadyCancelled == false;
+        }
+
+        @Override
+        public boolean isCancelled() {
+            return cancelled;
+        }
+
+        /**
+         * This method evaluates the health info of this node and if there is a change it sends an update request to the health node
+         */
+        @Override
+        public void run() {
+            if (cancelled) {
+                return;
+            }
+            boolean nextRunScheduled = false;
+            Runnable runOnceScheduleNextRunIfNecessary = new RunOnce(this::scheduleNextRunIfNecessary);
+            try {
+                ClusterState clusterState = clusterService.state();
+                HealthMetadata healthMetadata = HealthMetadata.getFromClusterState(clusterState);
+                if (healthMetadata != null) {
+                    DiskHealthInfo previousHealth = this.lastReportedDiskHealthInfo.get();
+                    DiskHealthInfo currentHealth = diskCheck.getHealth(healthMetadata, clusterState);
+                    if (currentHealth.equals(previousHealth) == false) {
+                        String nodeId = clusterService.localNode().getId();
+                        String healthNodeId = lastSeenHealthNode.get();
+                        ActionListener<AcknowledgedResponse> listener = ActionListener.wrap(response -> {
+                            // Update the last reported value only if the health node hasn't changed.
+                            if (Objects.equals(healthNodeId, lastSeenHealthNode.get())
+                                && lastReportedDiskHealthInfo.compareAndSet(previousHealth, currentHealth)) {
+                                logger.debug(
+                                    "Health info [{}] successfully sent, last reported value: {}.",
+                                    currentHealth,
+                                    lastReportedDiskHealthInfo.get()
+                                );
+                            }
+                        }, e -> {
+                            if (e.getCause() instanceof NodeNotConnectedException
+                                || e.getCause() instanceof HealthNodeNotDiscoveredException) {
+                                logger.warn("Failed to connect to the health node [{}], will try again.", e.getCause().getMessage());
+                            } else {
+                                logger.error(
```

Comment:
I see, yes, I do not have any strong objections. It might be difficult to catch it if something is going wrong here. But I can put it in debug and we will see if that is good enough in the longrun.